### ISSUE TITLE
View Resoler << yes Resoler

### ISSUE THREAD

**Issue by gregee123** [EcsRx/ecsrx.unity#29]
you have that typo in a few related files.

Side note: you react to the whole entity being removed, perhaps you should react similarly to the ViewComponent being removed from the entity.

**Comment by grofit** [comment]
Not noticed will have a look for the typo, but simple things like that by all means raise a PR or put a link in where the code is guff.

As far as ViewComponent being removed from an entity you can do that already by listening to the component removed event and adding a predicate `Where(x => x is ViewComponent)` or similar

**Comment by gregee123** [comment]
I now I can do it. Question is shouldn't it be done by ViewHandler for the same reason it subscribes to the entity being destroyed.

**Comment by grofit** [comment]
oh sorry yes I get what you mean, problem is you are very short with your blurbs so quite often I dont know if you are talking about a specific line of code in ecsrx, your own code which uses it, or just a high level approach being used somewhere, like if you had posted a link to the bit of code with your query:

https://github.com/grofit/ecsrx/blob/master/src/Assets/EcsRx/Unity/Systems/ViewHandler.cs#L67

> you react to the whole entity being removed, perhaps you should react similarly to the ViewComponent being removed from the entity.

I would have had a bit more context, I thought originally you were having problems yourself trying to listen to a view component being removed from an entity in your own code somewhere.

Anyway I think it can just be replaced with listening to the `ViewComponent` being removed for that entity and do the destruction then, as with the other change for remove component being fired for all components on a destroyed entity it should always be raised, so the new code would look something like:

```
            EventSystem.Receive<ComponentRemovedEvent>()
                .First(x => x.Component is ViewComponent && x.Entity == entity)
                .Subscribe(x =>
                {
                    if (viewSubscription != null)
                    { viewSubscription.Dispose(); }

                    DestroyView(viewObject);
                });
```

**Comment by gregee123** [comment]
I'll try to post more context next time. 

Two more things:

1. in the snippet above perhaps this subscription should also depend on DestroyWithView?

2. When you check if EntityView exist you should assign entity value even if that component already existed to err on the safe side otherwise you seem to be making an assumption that whoever added EntityView previously was linking the same entity to the GO.

**Comment by grofit** [comment]
Not too sure what you mean on the first point, the `DestroyWithView` var is there to indicate if the entity should be cleared up when the GO is destroyed (as in some cases you may want the entity to persist even if the associated GO has been destroyed).

The 2nd point is a tricky one, I would need to research how it currently works some more as I cannot remember off top of my head if there is some scenario where you may have a EntityView added (maybe from EntitySetup view) where you already have the view bound at this point, but if you were to have an entity bound elsewhere for some reason and then overwrote it with this one that could cause issues, so I think maybe this entire process needs more investigation.

**Comment by gregee123** [comment]
1. Question is should a GO always be destroyed when ViewComponent is removed.

2. I can't think of a scenario in which EntityView.entity should not == entity to which ViewComponent is being added. With the way it is currently it's by all means possible.

**Comment by grofit** [comment]
1. Simply speaking I would assume so, however if you are pooling it (via a pooling based system or your own implementation) that GO would not be destroyed and would just be put back into the pool.

2. I cannot think of a scenario off top of my head but that does not mean it wouldn't exist so its a tricky one to deal with, for the moment I think it is fine to keep it as is in this new version and then if there are issues with that use case then we address that when it occurs.

**Comment by gregee123** [comment]
1. problem is the GO is always destroyed.

**Comment by grofit** [comment]
It is when you are using a default `ViewHandler` and you are using a default `ViewResolverSystem` if you are using [ `PooledViewResolverSystem`](https://github.com/grofit/ecsrx/blob/03e48f5fd7eda2a47b96d10a65e8cba5210efb67/src/Assets/EcsRx/Unity/Systems/PooledViewResolverSystem.cs#L40) it doesnt use the `IViewHandler` by default and will just recycle the view, so it is up to the consumer if they want to destroy the GO when the view is removed, or use a pooling approach as provided, or they can implement their own system/view handler which would not destroy and do things however that user wants.

**Comment by gregee123** [comment]
1. Guess those default systems weren't meant to be flexible.

2. It would appear you're not exactly sure what EntityView is meant to do otherwise this wouldn't be a problem.

**Comment by grofit** [comment]
I do enjoy our exchanges, I would ask that you at least attempt to be a little nicer though, if you dislike any of my technical design or architecture you are free to fork it and do as you wish, I am in no way hindering your way of life or happiness levels, or if you wish to pay me for my time I am happy to be berated at an agreed daily fee.

So that being said:

1) They are flexible, they are also implementations of interfaces which can be replaced, if you are unwilling to replace them for your rather bespoke concerns (which you haven't even stipulated, just talked around some possible issue) you can always just fork it and make your changes with some nice happy message telling me you want me to spend some time to integrate it.

2) I am happy you have finally realised my general incompetence in all technical matters, however the problem is not "what it is" but "how it's used". So EntityView is merely there to say "this GO is bound to this entity", if you were in editor time you shouldnt be using it, you should be using `RegisterAsEntity` or whatever its called (there is a niche use case with a prefab which may use former). So I know exactly "what" it does, and for the most cases how its used, however I developer the library for my use cases (and hopefully most common ones for other people), so as unity is a bit of a mess and lets you quite often hang yourself without realising it, you could add `EntityView` onto anything you like and pre-set the entity through MBs or whatever, so as there *could* be a use case for people doing this I have not thought of its hard to tell you the unknown, if you have all of the answers please feel free to provide the use cases (preferably in Given, When, Then formats).

**Comment by grofit** [comment]
This has been released in 0.8.0 feel free to re-open if there are any issues with this
